spec/types/module/test-readonly.ts

```diff
@@ -0,0 +1,47 @@
+import { ReadonlyObservable, Observable, ReadonlyComputed} from "knockout";
+import * as ko from "knockout";
+
+function testReadonlyObservable() {
+    const write = ko.observable("foo");
+    write("bar");
+    const read = write as ReadonlyObservable<string>;
+
+    read(); // $ExpectType string
+    read.subscribe(() => {});  // Can still subscribe
+    // But can't write to it
+    // read("foo") // $ExpectError // Don't currently have a good mechanism for testing this outside of DT repo
```

Comment:
This file is pretty much copied from the `DefinitelyTyped` repo; unfortunately it's pretty hard to write a test that ensures a particular expression *doesn't* compile in TS.  They have some fairly sophisticated test mechanisms that allowed invalid lines to be annotated with this `//$ExpectError`  comment, and the test would fail if the line didn't error.

I couldn't find a good way to replicate that here, generally, so I just left the comments for posterity.